content/articles/polymorphism-in-java/index.md

```diff
@@ -0,0 +1,220 @@
+
+### Advantages of Polymorphism
+We are now going to look at the advantages of each of the types of polymorphism.
+
+**Dynamic Polymorphism**
+
+1. `Support for Method Overriding` - This is the feature where a child method can modify the specifications of the parent method. This also enables the creation of one method and modification to serve more than one method.
+2. `Method specification` - This allows for child classes to make modifications or improvements to the parent class before implementation.
+3. `Method Overloading` - This allows the use of the same method name on different methods but modifying the parameters and arguments to avoid a compile-time error.
+
+### Characteristics of Polymorphism
+
+1. *Polymorphic Coercion* - **Coercion** is the automatic conversion of data types from one form to another. An instance of the process mentioned is the conversion of a double into an integer. The above can also be reffed to as implicit type conversion.
+
+The example below will indicate how java achieves coercion.
+```java
+// Explicit type conversion
+double x;
+x=(double)2;
+```
+```java
+// Implicit type conversion
+double x;
+x=2;
+```
+In the first example, you first have to initialize the data type, and then when allocating the variable, you specify the type of the variable. The case is different in Implicit type conversion. After you initialize the data type, you only have to allocate the variable and the compiler will automatically recognize it as a double.
+
+2. *Operator Overloading* - This is where the same symbol or operator has more than one meaning even when used in the same code. Below is an example to explain operator overloading.
+```java
+String str = "5"+5;
+int sum = 5+5;
+
+System.out.println("str=%s sum=%d",str,sum);
+```
+
+```java
+//Output
+str = 55
+str = 10
+```
+
+In the above example, you can note that the `+` operator is not only used as a mathematical operator for addition, it is also used string concatenation by linking together a string and an integer.
+As seen in the output, the output is different even though the operator used is the same. This shows how Operator overloading happens.
+
+3. *Polymorphic parameters* - This particular characteristic allows a name of a parameter or method to be associated with different data types either locally or globally.
+
+```java
+public class Dog extends Animal {
+    private string legs;
+    public String setDogLegs(){
+        int legs = 4;
+        this.legs = this.legs + legs;
+    }
+}
+```
+
+Declaring of polymorphic parameters can lead to `variable hiding`.
+
+**Variable hiding** occurs where the declaration of a local variable overrides the global variable with the same name.
+
+In the above example, we have declared the legs globally as a string and then declared them locally as an integer. Normally variable hiding would have occurred. But to solve for that, we have used a global reference `this` to point to the global variables within the local context.
+
```

Comment:
Your article is missing a conclusion, add it after the summary. You may also add a further reading.

Reply:
You could also replace the conclusion with the summary.